docs/source/appendix.rst

```diff
@@ -14,12 +12,128 @@ notes apply:
 
-.. note::
-    The mode parameter controls how the update is performed. If you set 'patch', the call will only update
-    the fields in the Item object that are not None. This means that if the items exists from before, the
-    fields that are not specified will not be changed. If you set 'replace', all the fields that are not
-    specified, i.e., set to None and support being set to null, will be nulled out. See the API
-    documentation for the update endpoint for more information.
+.. _appendix-update:
+
+Update and Upsert Mode Parameter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The mode parameter controls how the update is performed. If you set 'patch', the call will only update
+the fields in the Item object that are not None. This means that if the items exists from before, the
+fields that are not specified will not be changed. The 'replace_ignore_null' works similarly, to
+'patch', but instead of updating it will replace the fields with new values. There is not difference
+between 'patch' and 'replace_ignore_null' for fields that only supports set. For example, `name` and
+`description` on TimeSeries. However, for fields that supports `set` and `add/remove`, like `metadata`,
+'patch` will add to the metadata, while 'replace_ignore_null' will replace the metadata with the new
+metadata. If you set 'replace', all the fields that are not specified, i.e., set to None and
+support being set to null, will be nulled out.
+
+Example **patch**:
+
+.. code:: python
+
+    >>> from cognite.client import CogniteClient
+    >>> client = CogniteClient()
+    >>>
+    >>> new_ts = client.time_series.create(
+    ...     TimeSeriesWrite(
+    ...         external_id="new_ts",
+    ...         name="New TS",
+    ...         metadata={"key": "value"}
+    ...     )
+    ... )
+    >>>
+    >>> updated = client.time_series.update(
+    ...     TimeSeriesWrite(
+    ...         external_id="new_ts",
+    ...         description="Updated description",
+    ...         metadata={"new": "entry"}
+    ...     ),
+    ...     mode="patch"
+    ... )
+    >>> print(updated.dump())
+    {'externalId': 'new_ts', 'name': 'New TS', 'description': 'Updated description', 'metadata': {'key': 'value', 'new': 'entry'}}
+
+Example **replace**:
+
+.. code:: python
+
+    >>> from cognite.client import CogniteClient
+    >>> client = CogniteClient()
+    >>>
+    >>> new_ts = client.time_series.create(
+    ...     TimeSeriesWrite(
+    ...         external_id="new_ts",
+    ...         name="New TS",
+    ...         metadata={"key": "value"}
+    ...     )
+    ... )
+    >>>
+    >>> updated = client.time_series.update(
+    ...     TimeSeriesWrite(
+    ...         external_id="new_ts",
+    ...         description="Updated description",
+    ...         metadata={"new": "entry"}
+    ...     ),
+    ...     mode="patch"
+    ... )
+    >>> print(updated.dump())
+    {'externalId': 'new_ts', 'description': 'Updated description', 'metadata': {'new': 'entry'}}
```

Comment:
Lets add a comment explicitly saying "name was not in update but still got removed"